src/content/en/updates/posts/2016/03/web-enabled-internet-of-things.markdown

```diff
@@ -0,0 +1,523 @@
+> chain will happen properly.
+
+Below we show a simplified version of my code, which will only work
+with the latest Web Bluetooth API, thus requiring Chrome Dev on Android.
+
+{% highlight javascript %}
+navigator.bluetooth.requestDevice({
+  filters: [{ name: 'Edison' }],
+  optionalServices: [0xFC00]
+})
+
+.then((device) => {
+  return device.gatt.connect();
+})
+
+.then((server) => {
+  return server.getPrimaryService(0xFC00));
+});
+
+.then((service) => {
+  let p1 = () => service.getCharacteristic(0xFC0B)
+  .then((characteristic) => {
+    this.colorLedCharacteristic = characteristic;
+    this.readLedColor();
+   });
+
+  let p2 = () => service.getCharacteristic(0xFC0A)
+  .then((characteristic) => {
+    characteristic.addEventListener(
+      'Characteristicvaluechanged', this.onTemperatureChange);
+    return characteristic.startNotifications();
+  });
+
+  return p1().then(p2);
+})
+
+.catch((err) => {
+  // Catch any error.
+})
+          
+.then(() => {
+  // Connection fully established, unless there was an error above.
+});
+{% endhighlight %}
+
+Reading and writing a string from a `DataView`/`ArrayBuffer` (what
+the WebBluetooth API uses) is just as easy as using `Buffer` on the
+Node.JS side, all we need to use is `TextEncoder` and `TextDecoder`:
+
+{% highlight javascript %}
+readLedColor: function() {
+  return this.colorLedCharacteristic.readValue()
+  .then(function(data) {
+    // In Chrome 50+, a DataView is returned instead of an ArrayBuffer.
+    data = data.buffer ? data : new DataView(data);
+    let decoder = new TextDecoder("utf-8");
+    let decodedString = decoder.decode(data);
+    document.querySelector('#color').value = decodedString;
+  });
+},
+
+writeLedColor: function() {
+  let encoder = new TextEncoder("utf-8");
+  let value = document.querySelector('#color').value;
+  let encodedString = encoder.encode(value.toLowerCase());
+
+  this.colorLedCharacteristic.writeValue(encodedString);
+},
+{% endhighlight %}
+
+Handling the characteristicvaluechanged event for the temperature
+sensor is also quite easy:
+   
+{% highlight javascript %}
+onTemperatureChange: function(event) {
+  let data = event.target.value;
+  // In Chrome 50+, a DataView is returned instead of an ArrayBuffer.
+  data = data.buffer ? data : new DataView(data);
+  let temperature = data.getFloat64(0, /*littleEndian=*/ true);
+  document.querySelector('#temp').innerHTML = temperature.toFixed(0);
+},
+{% endhighlight %}
+
+# Summary
+
+That was it folks! As you can see, communicating with Bluetooth Low
+Energy using Web Bluetooth on the client side and Node.JS on the
+Edison is quite easy and very powerful.
+
+Using the Physical Web and Web Bluetooth, Chrome will find the
+device and allow the user to easily connect to it, without installing
+a multitude of applications that will update from time to time even
+when the user seldom uses them.
+
+## Demo
+
+You can try of the [client](https://edison-webbt.appspot.com) to get
```

Comment:
`you can try of the client to get inspired` doesn't read right? what were you trying to say?